components/list/list-item.js

```diff
@@ -195,30 +203,55 @@ class ListItem extends LinkMixin(RtlMixin(LitElement)) {
 		fastdom.measure(() => this.resizedCallback(this.offsetWidth));
 	}
 
+	get checked() {
+		return this._checked;
+	}
+
+	set checked(value) {
+		const oldVal = this.checked;
+		this._checked = value;
+		this.requestUpdate('checked', oldVal).then(() => {
+			const checkBox = this.shadowRoot.querySelector('.d2l-list-item-checkbox');
+			checkBox && (checkBox.checked = value);
```

Comment:
So I had to set the checked property via here. 

I ran into problems putting it in the render control like it should be. The problems I ran into were before I added the `updated` control so I should try doing it the proper way again.

[] Try adding `?checked=${this.checked}`

Reply:
Yeah hopefully we can avoid the custom setters/getters.

Reply:
This one will still be required for this part:
```
if (typeof this.ref === 'undefined') {
	this.ref = ListItem.uniqueRef();
}
```

I will think about if this can be done in another life cycle event. It be nice to place it in `updated`

Reply:
I tried to add `?checked=${this.checked}` 

The issue is that the `:checked` css property doesn't change. I was able to remove the getter/setter after some refactoring on using proper life cycle methods from other required changes.

Reply:
I really want to get to the bottom of this, because having getters/setters for no apparent reason is troubling. I'm going to check out your branch and investigate.

Reply:
Wait, let me push up the new changes which will be easier to work with. The tests are broken and I am fixing them but still better to work with.

Reply:
I was able to get this working by adding `?checked="${this.selected}"` (there is no internal `checked` property on list items). Was that maybe all that was missing?

Reply:
Did the select all work? 

Were you able to remove 
```
const checkBox = this.shadowRoot.querySelector('.d2l-list-item-checkbox');
checkBox && (checkBox.checked = value);
```

I did do` ?checked="${this.selected}"` vs the `?checked="${this.checked}"` In my comment I copied the pervious comment and that is why it is missed placed.

Reply:
See comment: https://github.com/BrightspaceUI/core/pull/212#discussion_r325689455

Reply:
@dlockhart Found the issue. Use a `.` https://github.com/Polymer/lit-element/issues/601